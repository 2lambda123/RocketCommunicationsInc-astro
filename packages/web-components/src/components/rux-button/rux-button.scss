:host {
    /**
     * @prop --button-text-color: [DEPRECATED] Button text color
     */
    --button-text-color: var(--color-text-inverse);

    /**
     * @prop --button-background-color: [DEPRECATED] Button background color
     */
    --button-background-color: var(--color-background-interactive-default);

    /**
     * @prop --button-border-color: [DEPRECATED] Button border color
     */
    --button-border-color: transparent;

    /**
     * @prop --button-hover-background-color: [DEPRECATED] Button hover background color
     */
    --button-hover-background-color: var(--color-background-interactive-hover);

    /**
     * @prop --button-hover-border-color: [DEPRECATED] Button hover border color
     */
    --button-hover-border-color: transparent;

    /**
     * @prop --button-active-background-color: [DEPRECATED] Button active background color
     */
    --button-active-background-color: var(
        --color-background-interactive-default
    );

    /**
     * @prop --button-active-border-color: [DEPRECATED] Button active border color
     */
    --button-active-border-color: var(--color-background-interactive-default);

    /**
     * @prop --button-secondary-text-color: [DEPRECATED] Button secondary text color
     */

    --button-secondary-text-color: var(--color-background-interactive-default);

    /**
    * @prop --button-secondary-hover-text-color: [DEPRECATED] Secondary button hover text color
    */
    --button-secondary-hover-text-color: var(
        --color-background-interactive-hover
    );

    /**
     * @prop --button-secondary-background-color: [DEPRECATED] Button secondary background color
     */
    --button-secondary-background-color: transparent;

    /**
     * @prop --button-secondary-border-color: [DEPRECATED] Button secondary border color
     */
    --button-secondary-border-color: var(
        --color-background-interactive-default,
        rgb(0, 90, 143)
    );

    /**
     * @prop --button-secondary-hover-background-color: [DEPRECATED]  Button secondary hover background color
     */
    --button-secondary-hover-background-color: transparent;

    /**
     * @prop --button-secondary-hover-border-color: [DEPRECATED]  Button secondary hover border color
     */
    --button-secondary-hover-border-color: var(
        --color-background-interactive-hover
    );

    /**
    * @prop --button-borderless-text-color: [DEPRECATED] Button borderless text color
    */
    --button-borderless-color: var(--color-background-interactive-default);

    /**
    * @prop --button-borderless-hover-color: [DEPRECATED] Borderless buttons hover text color 
    */
    --button-borderless-hover-color: var(--color-background-interactive-hover);

    display: inline-block;
    width: auto;
}

:host([hidden]) {
    display: none;
}

:host([disabled]) {
    pointer-events: none;
}
.rux-button {
    display: flex;
    position: relative;
    margin: 0;
    width: inherit;
    padding: 0 1rem;
    height: 2.125rem;
    min-width: 2.25rem;
    border-radius: var(--radius-base);
    color: var(--button-text-color);
    font-family: var(--font-body-1-font-family);
    font-size: var(--font-body-1-font-size);
    font-weight: var(--font-body-1-font-weight);
    letter-spacing: var(--font-body-1-letter-spacing);
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    justify-content: center;
    align-items: center;
    user-select: none;

    rux-icon,
    ::slotted(rux-icon) {
        height: 1rem;
        width: 1rem;
        margin-right: 0.625rem;
        color: var(--button-text-color);
    }

    &:disabled {
        opacity: var(--disabled-opacity);
        cursor: not-allowed;
        pointer-events: none;
    }

    &:focus {
        outline: none;
    }

    &:hover:not([disabled]) {
        cursor: pointer;
    }

    &--default {
        border: 1px solid var(--button-border-color);
        background-color: var(--button-background-color);
        &:hover:not([disabled]) {
            border-color: var(--button-hover-border-color);
            background-color: var(--button-hover-background-color);
        }
        &:active:not([disabled]) {
            border-color: var(--button-active-border-color);
            background-color: var(--button-active-background-color);
        }
    }

    &--secondary {
        color: var(--button-secondary-text-color);
        background-color: var(--button-secondary-background-color);
        border: 1px solid var(--button-secondary-border-color);

        &:hover:not([disabled]) {
            color: var(--button-secondary-hover-text-color);
            background-color: var(--button-secondary-hover-background-color);
            border-color: var(--button-secondary-hover-border-color);
            rux-icon,
            ::slotted(rux-icon) {
                color: var(--button-secondary-hover-text-color);
            }
        }

        &:active:not([disabled]) {
            border-color: var(--button-secondary-border-color);
            background-color: var(--button-secondary-background-color);
            color: var(--button-secondary-text-color);
            rux-icon,
            ::slotted(rux-icon) {
                color: var(--button-secondary-text-color);
            }
        }
        rux-icon,
        ::slotted(rux-icon) {
            color: var(--button-secondary-text-color);
        }
    }
    &--borderless {
        color: var(--button-borderless-color);
        border: none;
        background: none;
        &:hover:not([disabled]) {
            color: var(--button-borderless-hover-color);
            background: none;
            rux-icon,
            ::slotted(rux-icon) {
                color: var(--button-borderless-hover-color);
            }
        }
        &:active:not([disabled]) {
            color: var(--button-borderless-color);
            rux-icon,
            ::slotted(rux-icon) {
                color: var(--button-borderless-color);
            }
        }
        rux-icon,
        ::slotted(rux-icon) {
            color: var(--button-borderless-color);
        }
    }

    &--small {
        height: 1.625rem;
        padding: 0 1rem;
        line-height: 1;
    }

    &--large {
        height: 2.875rem;
        min-width: 3rem;
        padding: 0 1rem;
    }

    &--icon-only {
        font-size: 0;
        width: 3rem;
    }
}

.rux-button--icon-only ::slotted(rux-icon),
.rux-button--icon-only rux-icon {
    margin-left: -0.625rem;
    margin-right: -0.625rem;
}
