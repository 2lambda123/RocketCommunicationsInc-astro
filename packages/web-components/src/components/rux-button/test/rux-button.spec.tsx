import { newSpecPage } from '@stencil/core/testing'
import { RuxButton } from '../rux-button'

describe('rux-button', () => {
    it('builds', async () => {
        const button = new RuxButton()

        expect(button).toBeTruthy()
        expect(button).toEqual({
            iconOnly: false,
            handleClick: expect.any(Function),
            secondary: false,
            borderless: false,
            disabled: false,
            type: 'button',
        })
    })

    it('renders', async () => {
        const page = await newSpecPage({
            components: [RuxButton],
            html: `<rux-button>Button</rux-button>`,
        })

        expect(page.root).toMatchSnapshot()
    })

    it('sets attributes', async () => {
        const page = await newSpecPage({
            components: [RuxButton],
            html: `<rux-button type="submit" secondary disabled>Button</rux-button>`,
        })

        expect(page.root).toMatchSnapshot()
    })
})
