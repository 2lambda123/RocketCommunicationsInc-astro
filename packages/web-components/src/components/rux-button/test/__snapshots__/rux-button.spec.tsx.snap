// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rux-button renders 1`] = `
<rux-button>
  <mock:shadow-root>
    <button class="rux-button rux-button--default" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
</rux-button>
`;

exports[`rux-button sets attributes 1`] = `
<rux-button disabled="" secondary="" type="submit">
  <mock:shadow-root>
    <button aria-disabled="true" class="rux-button rux-button--secondary" disabled="" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Button
</rux-button>
`;
